packages/debugger-extension/src/index.ts

```diff
@@ -166,19 +174,43 @@ const files: JupyterFrontEndPlugin<void> = {
 const notebooks: JupyterFrontEndPlugin<void> = {
   id: '@jupyterlab/debugger-extension:notebooks',
   autoStart: true,
-  requires: [IDebugger, INotebookTracker],
+  requires: [IDebugger, INotebookTracker, IMainMenu],
   optional: [ILabShell],
   activate: (
     app: JupyterFrontEnd,
     service: IDebugger,
     notebookTracker: INotebookTracker,
+    mainMenu: IMainMenu,
     labShell: ILabShell | null
   ) => {
+
     const handler = new Debugger.Handler({
       type: 'notebook',
       shell: app.shell,
       service
     });
+    
+    app.commands.addCommand('debugger:restart-debug', {
+      label: 'Restart and Debug',
+      isEnabled: () => { return true },
+      execute: async () => {
+        const state = service.getDebuggerState();
+        console.log(state.cells);
+        const { context, content } = notebookTracker.currentWidget!;
+
+        await service.stop();
+        const restarted = await sessionContextDialogs!.restart(context.sessionContext);
+        if (restarted) {
+          await service.restoreDebuggerState(state);
+          await handler._update(notebookTracker.currentWidget!,
+                                notebookTracker.currentWidget!.sessionContext.session);
```

Comment:
This is my main concern, I had to pass the `_update` method to public and call it after restoring the debugger state, otherwise the toggle button is switched off and the breakpoints are not restored in the UI. THey are correclty sent to the kernel though, since the code breaks as expected.

We don't want to call the publid method `update` since all the signal handlers for different channels and other stuffs restored in this method are already there. So I wonder if we should add a special entry for the `restart and debug` feature to the handler, or simply pass this method in public.